{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Requirements"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: google-cloud-bigquery in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (3.17.2)\n",
      "Requirement already satisfied: db-dtypes in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (1.2.0)\n",
      "Requirement already satisfied: google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from google-cloud-bigquery) (2.15.0)\n",
      "Requirement already satisfied: google-cloud-core<3.0.0dev,>=1.6.0 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from google-cloud-bigquery) (2.4.1)\n",
      "Requirement already satisfied: google-resumable-media<3.0dev,>=0.6.0 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from google-cloud-bigquery) (2.7.0)\n",
      "Requirement already satisfied: packaging>=20.0.0 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from google-cloud-bigquery) (23.2)\n",
      "Requirement already satisfied: python-dateutil<3.0dev,>=2.7.2 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from google-cloud-bigquery) (2.8.2)\n",
      "Requirement already satisfied: requests<3.0.0dev,>=2.21.0 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from google-cloud-bigquery) (2.31.0)\n",
      "Requirement already satisfied: pandas>=0.24.2 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from db-dtypes) (2.2.0)\n",
      "Requirement already satisfied: pyarrow>=3.0.0 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from db-dtypes) (15.0.0)\n",
      "Requirement already satisfied: numpy>=1.16.6 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from db-dtypes) (1.26.3)\n",
      "Requirement already satisfied: googleapis-common-protos<2.0.dev0,>=1.56.2 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->google-cloud-bigquery) (1.62.0)\n",
      "Requirement already satisfied: protobuf!=3.20.0,!=3.20.1,!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0.dev0,>=3.19.5 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->google-cloud-bigquery) (4.25.2)\n",
      "Requirement already satisfied: google-auth<3.0.dev0,>=2.14.1 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->google-cloud-bigquery) (2.26.2)\n",
      "Requirement already satisfied: google-crc32c<2.0dev,>=1.0 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from google-resumable-media<3.0dev,>=0.6.0->google-cloud-bigquery) (1.5.0)\n",
      "Requirement already satisfied: pytz>=2020.1 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from pandas>=0.24.2->db-dtypes) (2023.3.post1)\n",
      "Requirement already satisfied: tzdata>=2022.7 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from pandas>=0.24.2->db-dtypes) (2023.4)\n",
      "Requirement already satisfied: six>=1.5 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from python-dateutil<3.0dev,>=2.7.2->google-cloud-bigquery) (1.16.0)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from requests<3.0.0dev,>=2.21.0->google-cloud-bigquery) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from requests<3.0.0dev,>=2.21.0->google-cloud-bigquery) (3.6)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from requests<3.0.0dev,>=2.21.0->google-cloud-bigquery) (2.1.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from requests<3.0.0dev,>=2.21.0->google-cloud-bigquery) (2023.11.17)\n",
      "Requirement already satisfied: cachetools<6.0,>=2.0.0 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from google-auth<3.0.dev0,>=2.14.1->google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->google-cloud-bigquery) (5.3.2)\n",
      "Requirement already satisfied: pyasn1-modules>=0.2.1 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from google-auth<3.0.dev0,>=2.14.1->google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->google-cloud-bigquery) (0.3.0)\n",
      "Requirement already satisfied: rsa<5,>=3.1.4 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from google-auth<3.0.dev0,>=2.14.1->google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->google-cloud-bigquery) (4.9)\n",
      "Requirement already satisfied: pyasn1<0.6.0,>=0.4.6 in /home/spritan/.conda/envs/Hestia2/lib/python3.11/site-packages (from pyasn1-modules>=0.2.1->google-auth<3.0.dev0,>=2.14.1->google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->google-cloud-bigquery) (0.5.1)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "%pip install --upgrade google-cloud-bigquery db-dtypes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from pprint import pprint\n",
    "\n",
    "import pandas as pd\n",
    "from google.cloud import bigquery\n",
    "\n",
    "from face_cluser import FaceCluster"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = '/run/media/spritan/New Volume/Github/Google_Solution/Hestia_latest/HESTIA/notebooks/keys/detectionKey.json'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Init"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:google.auth._default:Checking /run/media/spritan/New Volume/Github/Google_Solution/Hestia_latest/HESTIA/notebooks/keys/detectionKey.json for explicit credentials as part of auth process...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:google.auth._default:Checking /run/media/spritan/New Volume/Github/Google_Solution/Hestia_latest/HESTIA/notebooks/keys/detectionKey.json for explicit credentials as part of auth process...\n"
     ]
    }
   ],
   "source": [
    "client = bigquery.Client()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### retrival"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "sql_query = \"\"\"\n",
    "SELECT * FROM `plenary-truck-411220.clustering_dataset.faces`\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:urllib3.connectionpool:https://bigquery.googleapis.com:443 \"POST /bigquery/v2/projects/plenary-truck-411220/jobs?prettyPrint=false HTTP/1.1\" 200 None\n",
      "DEBUG:urllib3.connectionpool:https://bigquery.googleapis.com:443 \"GET /bigquery/v2/projects/plenary-truck-411220/queries/75a489da-4591-4c02-ae4d-39cfd53f1d8d?maxResults=0&location=asia-south1&prettyPrint=false HTTP/1.1\" 200 None\n",
      "DEBUG:urllib3.connectionpool:https://bigquery.googleapis.com:443 \"GET /bigquery/v2/projects/plenary-truck-411220/queries/75a489da-4591-4c02-ae4d-39cfd53f1d8d?fields=jobReference%2CtotalRows%2CpageToken%2Crows&location=asia-south1&formatOptions.useInt64Timestamp=True&prettyPrint=false HTTP/1.1\" 200 None\n"
     ]
    }
   ],
   "source": [
    "df = client.query(sql_query).result().to_dataframe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = FaceCluster()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:Utils.loggerHelper:<class 'pandas.core.frame.DataFrame'>\n",
      "INFO:Utils.loggerHelper:Quantifying faces...\n",
      "DEBUG:Utils.loggerHelper:Images loaded: 68\n",
      "DEBUG:Utils.loggerHelper:Processing image 1/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1704701781090_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1704701781090_0.png HTTP/1.1\" 200 2931382\n",
      "CRITICAL:Utils.loggerHelper:An error occurred: Error while calling cudaMalloc(&data, new_size*sizeof(float)) in file /home/conda/feedstock_root/build_artifacts/dlib-split_1704932578544/work/dlib/cuda/gpu_data.cpp:221. code: 2, reason: out of memory\n",
      "DEBUG:Utils.loggerHelper:Processing image 2/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1704701781090_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1704701781090_1.png HTTP/1.1\" 200 2446952\n",
      "DEBUG:Utils.loggerHelper:Processing image 3/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1704701781090_2.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1704701781090_2.png HTTP/1.1\" 200 827963\n",
      "DEBUG:Utils.loggerHelper:Processing image 4/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1704701781090_3.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1704701781090_3.png HTTP/1.1\" 200 4869005\n",
      "CRITICAL:Utils.loggerHelper:An error occurred: Error while calling cudaMalloc(&data, new_size*sizeof(float)) in file /home/conda/feedstock_root/build_artifacts/dlib-split_1704932578544/work/dlib/cuda/gpu_data.cpp:221. code: 2, reason: out of memory\n",
      "DEBUG:Utils.loggerHelper:Processing image 5/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705931867687_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1705931867687_0.png HTTP/1.1\" 200 5743406\n",
      "CRITICAL:Utils.loggerHelper:An error occurred: Error while calling cudaMalloc(&data, new_size*sizeof(float)) in file /home/conda/feedstock_root/build_artifacts/dlib-split_1704932578544/work/dlib/cuda/gpu_data.cpp:221. code: 2, reason: out of memory\n",
      "DEBUG:Utils.loggerHelper:Processing image 6/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705931867687_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1705931867687_1.png HTTP/1.1\" 200 2217709\n",
      "DEBUG:Utils.loggerHelper:Processing image 7/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705931867687_2.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1705931867687_2.png HTTP/1.1\" 200 3161343\n",
      "CRITICAL:Utils.loggerHelper:An error occurred: Error while calling cudaMalloc(&data, new_size*sizeof(float)) in file /home/conda/feedstock_root/build_artifacts/dlib-split_1704932578544/work/dlib/cuda/gpu_data.cpp:221. code: 2, reason: out of memory\n",
      "DEBUG:Utils.loggerHelper:Processing image 8/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605197792_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605197792_0.png HTTP/1.1\" 200 152840\n",
      "DEBUG:Utils.loggerHelper:Processing image 9/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605197792_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605197792_1.png HTTP/1.1\" 200 52713\n",
      "DEBUG:Utils.loggerHelper:Processing image 10/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605197792_2.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605197792_2.png HTTP/1.1\" 200 94050\n",
      "DEBUG:Utils.loggerHelper:Processing image 11/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605197792_3.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605197792_3.png HTTP/1.1\" 200 29791\n",
      "DEBUG:Utils.loggerHelper:Processing image 12/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605249930_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605249930_0.png HTTP/1.1\" 200 47982\n",
      "DEBUG:Utils.loggerHelper:Processing image 13/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605249930_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605249930_1.png HTTP/1.1\" 200 135335\n",
      "DEBUG:Utils.loggerHelper:Processing image 14/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605305009_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605305009_0.png HTTP/1.1\" 200 68749\n",
      "DEBUG:Utils.loggerHelper:Processing image 15/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605305009_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605305009_1.png HTTP/1.1\" 200 59162\n",
      "DEBUG:Utils.loggerHelper:Processing image 16/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605305009_2.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605305009_2.png HTTP/1.1\" 200 35828\n",
      "DEBUG:Utils.loggerHelper:Processing image 17/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605355095_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605355095_0.png HTTP/1.1\" 200 119403\n",
      "DEBUG:Utils.loggerHelper:Processing image 18/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605355095_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605355095_1.png HTTP/1.1\" 200 169713\n",
      "DEBUG:Utils.loggerHelper:Processing image 19/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605355095_2.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605355095_2.png HTTP/1.1\" 200 130436\n",
      "DEBUG:Utils.loggerHelper:Processing image 20/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605355095_3.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605355095_3.png HTTP/1.1\" 200 119192\n",
      "DEBUG:Utils.loggerHelper:Processing image 21/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605355095_4.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706605355095_4.png HTTP/1.1\" 200 51598\n",
      "DEBUG:Utils.loggerHelper:Processing image 22/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706607863159_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706607863159_0.png HTTP/1.1\" 200 47982\n",
      "DEBUG:Utils.loggerHelper:Processing image 23/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706607863159_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706607863159_1.png HTTP/1.1\" 200 135335\n",
      "DEBUG:Utils.loggerHelper:Processing image 24/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706607904499_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706607904499_0.png HTTP/1.1\" 200 119403\n",
      "DEBUG:Utils.loggerHelper:Processing image 25/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706607904499_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706607904499_1.png HTTP/1.1\" 200 169713\n",
      "DEBUG:Utils.loggerHelper:Processing image 26/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706607904499_2.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706607904499_2.png HTTP/1.1\" 200 130436\n",
      "DEBUG:Utils.loggerHelper:Processing image 27/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706607904499_3.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706607904499_3.png HTTP/1.1\" 200 119192\n",
      "DEBUG:Utils.loggerHelper:Processing image 28/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706607904499_4.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706607904499_4.png HTTP/1.1\" 200 51598\n",
      "DEBUG:Utils.loggerHelper:Processing image 29/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706774076366_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1706774076366_0.png HTTP/1.1\" 200 14106473\n",
      "CRITICAL:Utils.loggerHelper:An error occurred: Error while calling cudaMalloc(&data, new_size*sizeof(float)) in file /home/conda/feedstock_root/build_artifacts/dlib-split_1704932578544/work/dlib/cuda/gpu_data.cpp:221. code: 2, reason: out of memory\n",
      "DEBUG:Utils.loggerHelper:Processing image 30/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1707243976161_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1707243976161_0.png HTTP/1.1\" 200 11450855\n",
      "CRITICAL:Utils.loggerHelper:An error occurred: Error while calling cudaMalloc(&data, new_size*sizeof(float)) in file /home/conda/feedstock_root/build_artifacts/dlib-split_1704932578544/work/dlib/cuda/gpu_data.cpp:221. code: 2, reason: out of memory\n",
      "DEBUG:Utils.loggerHelper:Processing image 31/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1707580020676_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1707580020676_0.png HTTP/1.1\" 200 2739432\n",
      "CRITICAL:Utils.loggerHelper:An error occurred: Error while calling cudaMalloc(&data, new_size*sizeof(float)) in file /home/conda/feedstock_root/build_artifacts/dlib-split_1704932578544/work/dlib/cuda/gpu_data.cpp:221. code: 2, reason: out of memory\n",
      "DEBUG:Utils.loggerHelper:Processing image 32/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1707580020676_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1707580020676_1.png HTTP/1.1\" 200 1458724\n",
      "CRITICAL:Utils.loggerHelper:An error occurred: Error while calling cudaMalloc(&data, n) in file /home/conda/feedstock_root/build_artifacts/dlib-split_1704932578544/work/dlib/cuda/cuda_data_ptr.cpp:58. code: 2, reason: out of memory\n",
      "DEBUG:Utils.loggerHelper:Processing image 33/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1707580020676_2.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1707580020676_2.png HTTP/1.1\" 200 1709585\n",
      "DEBUG:Utils.loggerHelper:Processing image 34/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1707580121416_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1707580121416_0.png HTTP/1.1\" 200 4118416\n",
      "DEBUG:Utils.loggerHelper:Processing image 35/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1707580121416_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1707580121416_1.png HTTP/1.1\" 200 893246\n",
      "DEBUG:Utils.loggerHelper:Processing image 36/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1707580121416_2.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1707580121416_2.png HTTP/1.1\" 200 547486\n",
      "DEBUG:Utils.loggerHelper:Processing image 37/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1707580121416_3.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1707580121416_3.png HTTP/1.1\" 200 456220\n",
      "DEBUG:Utils.loggerHelper:Processing image 38/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1707580121416_4.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1707580121416_4.png HTTP/1.1\" 200 1032194\n",
      "DEBUG:Utils.loggerHelper:Processing image 39/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1707580121416_5.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1707580121416_5.png HTTP/1.1\" 200 23011\n",
      "DEBUG:Utils.loggerHelper:Processing image 40/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703882912473_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703882912473_0.png HTTP/1.1\" 200 118430\n",
      "DEBUG:Utils.loggerHelper:Processing image 41/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703749145245_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703749145245_0.png HTTP/1.1\" 200 293108\n",
      "DEBUG:Utils.loggerHelper:Processing image 42/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703749145245_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703749145245_1.png HTTP/1.1\" 200 366327\n",
      "DEBUG:Utils.loggerHelper:Processing image 43/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703749145245_2.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703749145245_2.png HTTP/1.1\" 200 300380\n",
      "DEBUG:Utils.loggerHelper:Processing image 44/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703749145245_3.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703749145245_3.png HTTP/1.1\" 200 198817\n",
      "DEBUG:Utils.loggerHelper:Processing image 45/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703711347601_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703711347601_0.png HTTP/1.1\" 200 2097630\n",
      "DEBUG:Utils.loggerHelper:Processing image 46/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703711347601_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703711347601_1.png HTTP/1.1\" 200 2356263\n",
      "DEBUG:Utils.loggerHelper:Processing image 47/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705263539822_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1705263539822_0.png HTTP/1.1\" 200 89437\n",
      "DEBUG:Utils.loggerHelper:Processing image 48/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705263539822_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1705263539822_1.png HTTP/1.1\" 200 28184\n",
      "DEBUG:Utils.loggerHelper:Processing image 49/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705263539822_2.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1705263539822_2.png HTTP/1.1\" 200 39340\n",
      "DEBUG:Utils.loggerHelper:Processing image 50/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705263539822_3.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1705263539822_3.png HTTP/1.1\" 200 92363\n",
      "DEBUG:Utils.loggerHelper:Processing image 51/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703935816635_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703935816635_0.png HTTP/1.1\" 200 1096271\n",
      "CRITICAL:Utils.loggerHelper:An error occurred: Error while calling cudaMalloc(&data, n) in file /home/conda/feedstock_root/build_artifacts/dlib-split_1704932578544/work/dlib/cuda/cuda_data_ptr.cpp:58. code: 2, reason: out of memory\n",
      "DEBUG:Utils.loggerHelper:Processing image 52/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703935816635_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703935816635_1.png HTTP/1.1\" 200 109798\n",
      "DEBUG:Utils.loggerHelper:Processing image 53/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703935816635_2.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703935816635_2.png HTTP/1.1\" 200 67967\n",
      "DEBUG:Utils.loggerHelper:Processing image 54/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703059580160_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703059580160_0.png HTTP/1.1\" 200 1110236\n",
      "DEBUG:Utils.loggerHelper:Processing image 55/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703946167111_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703946167111_0.png HTTP/1.1\" 200 480021\n",
      "DEBUG:Utils.loggerHelper:Processing image 56/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703946167111_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703946167111_1.png HTTP/1.1\" 200 123037\n",
      "DEBUG:Utils.loggerHelper:Processing image 57/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703946167111_2.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703946167111_2.png HTTP/1.1\" 200 72544\n",
      "DEBUG:Utils.loggerHelper:Processing image 58/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703946167111_3.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703946167111_3.png HTTP/1.1\" 200 149432\n",
      "DEBUG:Utils.loggerHelper:Processing image 59/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703946167111_4.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703946167111_4.png HTTP/1.1\" 200 120954\n",
      "DEBUG:Utils.loggerHelper:Processing image 60/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705499593272_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1705499593272_0.png HTTP/1.1\" 200 330071\n",
      "DEBUG:Utils.loggerHelper:Processing image 61/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705499593272_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1705499593272_1.png HTTP/1.1\" 200 185103\n",
      "DEBUG:Utils.loggerHelper:Processing image 62/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703706346466_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703706346466_0.png HTTP/1.1\" 200 1861023\n",
      "DEBUG:Utils.loggerHelper:Processing image 63/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703706346466_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703706346466_1.png HTTP/1.1\" 200 1225803\n",
      "CRITICAL:Utils.loggerHelper:An error occurred: Error while calling cudaMalloc(&data, n) in file /home/conda/feedstock_root/build_artifacts/dlib-split_1704932578544/work/dlib/cuda/cuda_data_ptr.cpp:58. code: 2, reason: out of memory\n",
      "DEBUG:Utils.loggerHelper:Processing image 64/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703706346466_2.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703706346466_2.png HTTP/1.1\" 200 1086334\n",
      "CRITICAL:Utils.loggerHelper:An error occurred: Error while calling cudaMalloc(&data, n) in file /home/conda/feedstock_root/build_artifacts/dlib-split_1704932578544/work/dlib/cuda/cuda_data_ptr.cpp:58. code: 2, reason: out of memory\n",
      "DEBUG:Utils.loggerHelper:Processing image 65/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703706346466_3.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1703706346466_3.png HTTP/1.1\" 200 300825\n",
      "DEBUG:Utils.loggerHelper:Processing image 66/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705263475803_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1705263475803_0.png HTTP/1.1\" 200 409086\n",
      "DEBUG:Utils.loggerHelper:Processing image 67/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705263475803_1.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1705263475803_1.png HTTP/1.1\" 200 137462\n",
      "DEBUG:Utils.loggerHelper:Processing image 68/68\n",
      "DEBUG:Utils.loggerHelper:https://storage.googleapis.com/faces_for_clusters/CroppedImages/1702840841383_0.png\n",
      "DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): storage.googleapis.com:443\n",
      "DEBUG:urllib3.connectionpool:https://storage.googleapis.com:443 \"GET /faces_for_clusters/CroppedImages/1702840841383_0.png HTTP/1.1\" 200 1167045\n",
      "CRITICAL:Utils.loggerHelper:An error occurred: Error while calling cudaMalloc(&data, n) in file /home/conda/feedstock_root/build_artifacts/dlib-split_1704932578544/work/dlib/cuda/cuda_data_ptr.cpp:58. code: 2, reason: out of memory\n",
      "DEBUG:Utils.loggerHelper:serializing encodings...\n",
      "INFO:Utils.loggerHelper:Received df dataset.\n",
      "INFO:Utils.loggerHelper:Loading encodings...\n",
      "INFO:Utils.loggerHelper:Clustering ...\n",
      "INFO:Utils.loggerHelper:# unique faces: 1\n",
      "INFO:Utils.loggerHelper:Faces for face ID: -1\n",
      "DEBUG:Utils.loggerHelper:[-1 -1 -1 -1  0  0 -1  0  0 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1]\n",
      "[ WARN:0@316.126] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705499593272_0.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.126] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705931867687_1.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.126] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703706346466_0.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.126] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703711347601_0.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.126] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703711347601_1.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.126] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703059580160_0.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.126] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705263475803_1.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.127] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705499593272_1.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.127] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705263475803_0.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.127] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705263539822_0.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.127] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706607863159_1.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.127] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605197792_0.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.127] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1703749145245_3.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.127] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1707580020676_2.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.127] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605249930_1.png'): can't open/read file: check file path/integrity\n",
      "INFO:Utils.loggerHelper:Faces for face ID: 0\n",
      "[ WARN:0@316.127] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605305009_1.png'): can't open/read file: check file path/integrity\n",
      "DEBUG:Utils.loggerHelper:[-1 -1 -1 -1  0  0 -1  0  0 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1]\n",
      "[ WARN:0@316.128] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605355095_2.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.129] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1705263539822_3.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.129] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706607904499_2.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.129] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706605355095_0.png'): can't open/read file: check file path/integrity\n",
      "[ WARN:0@316.129] global loadsave.cpp:248 findDecoder imread_('https://storage.googleapis.com/faces_for_clusters/CroppedImages/1706607904499_0.png'): can't open/read file: check file path/integrity\n"
     ]
    }
   ],
   "source": [
    "res = data.getClusters(dataset=df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'-1': ['1705499593272_0',\n",
       "  '1705931867687_1',\n",
       "  '1703706346466_0',\n",
       "  '1703711347601_0',\n",
       "  '1703711347601_1',\n",
       "  '1703059580160_0',\n",
       "  '1705263475803_1',\n",
       "  '1705499593272_1',\n",
       "  '1705263475803_0',\n",
       "  '1705263539822_0',\n",
       "  '1706607863159_1',\n",
       "  '1706605197792_0',\n",
       "  '1703749145245_3',\n",
       "  '1707580020676_2',\n",
       "  '1706605249930_1',\n",
       "  '1706605305009_1'],\n",
       " '0': ['1706605355095_2',\n",
       "  '1705263539822_3',\n",
       "  '1706607904499_2',\n",
       "  '1706605355095_0',\n",
       "  '1706607904499_0']}"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Hestia2",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
